### PR TITLE
Proline support dda module issue #354

### PR DESCRIPTION
**Pull request by mlocardpaulet**
I started fixing ProlineStudio support. 
I still need to:
- [x] add link to json export template in the documentation
- [x] fix this bug in the parameter parsing when I want to do public upload: "UnicodeDecodeError: 'utf-8' codec can't decode byte 0xca in position 17: invalid continuation byte" 
- [x] add information on ProlineStudio version in documentation (what version started exporting the version in the parameters?)
- [x] add the correct files for testing

### CODE REVIEW

**Comment by mlocardpaulet** [comment]
@enryH, I know that you worked on this, right? 
image

Can I help? Is there something I miss?

**Comment by mlocardpaulet** [comment]
Also, test won't work since the toml and headers changed. @enryH: Shall I just change the test file? Or is there anything I should be aware of when I do that?

**Comment by enryH** [comment]
> @enryH, I know that you worked on this, right? image
> 
> Can I help? Is there something I miss?

It was related to reading an excel file. I will have to check if my changes were reversed somehow. But the basic problem is that the temporary file was dumped, and then it was opened as if it was a text file (always) - which is not true for excel files (binary).

https://github.com/Proteobench/ProteoBench/blob/456592b65597e84f38ac23e3dc25f9bbbc720b8b/webinterface/pages/base_pages/quant.py#L484-L488

But this looks still correct.

**Comment by mlocardpaulet** [comment]
> The error in the test module is related to column names
> 
> ```shell
> FAILED test/test_module_dda_quant.py::TestOutputFileReading::test_input_file_initial_parsing - KeyError: "The following id_vars or value_vars are not present in the DataFrame: ['abundance_LFQ_Orbitrap_DDA_Condition_A_Sample_Alpha_01.mgf', 'abundance_LFQ_Orbitrap_DDA_Condition_A_Sample_Alpha_02.mgf', 'abundance_LFQ_Orbitrap_DDA_Condition_A_Sample_Alpha_03.mgf', 'abundance_LFQ_Orbitrap_DDA_Condition_B_Sample_Alpha_01.mgf', 'abundance_LFQ_Orbitrap_DDA_Condition_B_Sample_Alpha_02.mgf', 'abundance_LFQ_Orbitrap_DDA_Condition_B_Sample_Alpha_03.mgf']"
> FAILED test/test_module_dda_quant.py::TestOutputFileReading::test_input_file_processing - KeyError: "The following id_vars or value_vars are not present in the DataFrame: ['abundance_LFQ_Orbitrap_DDA_Condition_A_Sample_Alpha_01.mgf', 'abundance_LFQ_Orbitrap_DDA_Condition_A_Sample_Alpha_02.mgf', 'abundance_LFQ_Orbitrap_DDA_Condition_A_Sample_Alpha_03.mgf', 'abundance_LFQ_Orbitrap_DDA_Condition_B_Sample_Alpha_01.mgf', 'abundance_LFQ_Orbitrap_DDA_Condition_B_Sample_Alpha_02.mgf', 'abundance_LFQ_Orbitrap_DDA_Condition_B_Sample_Alpha_03.mgf']"
> ```

I know, I need to make the test match to the new column names.

File: proteobench/io/parsing/io_parse_settings/Quant/lfq/ion/DDA/parse_settings_proline.toml [hunk 1]
```diff
@@ -6,20 +6,20 @@
...
+"abundance_LFQ_Orbitrap_DDA_Condition_B_Sample_Alpha_03.mgf" = "B"
 
 [run_mapper]
-"abundance_DDA_Condition_A_Sample_Alpha_01" = "Condition_A_Sample_Alpha_01"
-"abundance_DDA_Condition_A_Sample_Alpha_02" = "Condition_A_Sample_Alpha_02"
-"abundance_DDA_Condition_A_Sample_Alpha_03" = "Condition_A_Sample_Alpha_03"
-"abundance_DDA_Condition_B_Sample_Alpha_01" = "Condition_B_Sample_Alpha_01"
-"abundance_DDA_Condition_B_Sample_Alpha_02" = "Condition_B_Sample_Alpha_02"
-"abundance_DDA_Condition_B_Sample_Alpha_03" = "Condition_B_Sample_Alpha_03"
+"abundance_LFQ_Orbitrap_DDA_Condition_A_Sample_Alpha_01.mgf" = "Condition_A_Sample_Alpha_01"
```
Comment (enryH) [inline proteobench/io/parsing/io_parse_settings/Quant/lfq/ion/DDA/parse_settings_proline.toml:17]: If you change the names here, you will need to change it in the test data as well. this setting is run specific? Would probably be good to have it in a separate toml file for different runs.

Comment (mlocardpaulet) [reply proteobench/io/parsing/io_parse_settings/Quant/lfq/ion/DDA/parse_settings_proline.toml:17]: I am not sure that I understand what you want to do. "separate toml file for different runs"?

Comment (mlocardpaulet) [reply proteobench/io/parsing/io_parse_settings/Quant/lfq/ion/DDA/parse_settings_proline.toml:17]: done!

**Comment by mlocardpaulet** [comment]
@enryH I made the changes in the test file (quant). It should be good to go. I still need to add the template link in the documentation but for this I think that I need Julian's help. 
And there is still this utf-8 error when we send the PR for public submission

**Comment by mlocardpaulet** [comment]
I noticed a bug in the parsing of the quant results from ProlineStudio:
https://github.com/Proteobench/ProteoBench/issues/473
It needs to be fixed before we merge this PR.

**Comment by mlocardpaulet** [comment]
OK, @RobbinBouwmeester and @enryH, I think that it is all good. 
I am sorry I also changed the logos in this PR, not the cleanest. I added a few missing institutes and the core 4 life logo. 

Basically, here you'll find my changes to fix the ProlineStudio column header naming, and precursor redundancy issue (I basically copied what was done by Robbin in the past but was lost somehow - it is in `proteobench/io/parsing/parse_ion.py`). I also adapted the .toml. It seems to work when I run it locally.
@enryH fixed an error that occurred when submitting publicly (the UnicodeDecodeError: 'UTF-8' mentioned in a comment above).
I hope that this is clear enough. And I don't know why the formatting checks all fail... Happy to learn why if you have time to explain. In general, any comment is more than welcome. 
If it is OK, I can move on and try to add MSAngel compatibility (it is basically the same output formats, and for the parameter parsing I'll see in a second step).

**Pull request merged**